src/main/kotlin/com/openlattice/shuttle/Shuttle.kt

```diff
@@ -95,15 +116,58 @@ class Shuttle(
         }
     }
 
-    private val uploadRegulator = Semaphore(threads)
+    //vars to be used only when shuttle is run on shuttle server
+    private var writeLog: (String, String, OffsetDateTime, IntegrationStatus) -> Unit
+    private lateinit var logEntitySet: EntitySet
+    private lateinit var logsDestination: PostgresDestination
+    private lateinit var integrationJobs: IMap<UUID, IntegrationJob>
+    private val logProperties = mutableMapOf<FullQualifiedName, PropertyType>()
+    private val ptidsByBlackboxProperty = mutableMapOf<BlackboxProperty, UUID>()
+
+    init {
+        if (blackbox.enabled) {
+            val jobId = maybeJobId.get()
+            integrationJobs = HazelcastMap.INTEGRATION_JOBS.getMap( hazelcastInstance!! )
+
+            this.writeLog = { name, log, time, status ->
+                logger.info(log)
+                storeLog(name, log, time, status, jobId)
+                integrationJobs.executeOnKey(jobId, UpdateIntegrationStatusEntryProcessor(status))
+            }
+
+            blackbox.fqns.forEach {
+                val fqn = FullQualifiedName(it.value)
+                val propertyType = propertyTypes.getValue(fqn)
```

Comment:
we might want to cache these to a local map to avoid reading all the property types from hazelcast every time something is logged (if we feel safe assuming the blackbox fqns won't change while shuttle server is running). If not, we should at least switch to a propertyTypes.getAll() to avoid the extra hazelcast latency

Reply:
loool that is what you are doing here, nvm